cmd/publishing-bot/publisher.go

```diff
@@ -65,10 +67,27 @@ func (p *PublisherMunger) updateSourceRepo() (string, error) {
 		return "", fmt.Errorf("failed running %v on %q repo: %v", strings.Join(cmd.Args, " "), p.config.SourceRepo, err)
 	}
 
-	rules, err := config.LoadRules(p.config.RulesFile)
+	var rules *config.RepositoryRules
+
+	if strings.HasPrefix(p.config.RulesFile, "/") {
+		// The rules are specified as absolute file path
+		rules, err = config.LoadRulesFromFile(p.config.RulesFile)
+	} else {
+		// The rules are specified as URL
+		if ruleUrl, err := url.ParseRequestURI(p.config.RulesFile); err == nil && len(ruleUrl.Host) > 0 {
+			rules, err = config.LoadRulesFromURL(ruleUrl)
+		} else {
+			// The rules are specified as relative path in repository dir
+			rules, err = config.LoadRulesFromRepo(p.config.RulesFile, repoDir, sourceBranch)
+		}
+	}
 	if err != nil {
 		return "", err
 	}
+	if rules == nil {
+		return "", fmt.Errorf("Unable to load rule file from %q", p.config.RulesFile)
+	}
+
 	p.reposRules = *rules
```

Comment:
why do we read the rules here and not in the main loop? Overriding the rules in `p` is ugly, now that they actually differ per branch.